### PR TITLE
melt relation primops

### PR DESCRIPTION
**Pull request by tranma**
....so we can compare things like `OptionT IntT`

might need a REPL test to demonstrate.
@amosr @jystic

### CODE REVIEW

File: src/Icicle/Avalanche/Statement/Simp/Constructor.hs [hunk 1]
```diff
@@ -234,35 +242,75 @@ constructor a_fresh statements
...
 
    | (PrimMinimal (Min.PrimRelation Min.PrimRelationNe t), [nx, ny]) <- prima
-   , Just ts <- tryMeltType t
-   , tis     <- List.zip ts [0..]
-   = Just $ primFold1 primOr
-   $ fmap (\(tv, i) -> primNe tv (primUnpack i t nx) (primUnpack i t ny)) tis
+   , Just tis <- withIndex tryMeltType t
+   = Just
+   $ primFold1 primOr
+   $ fmap (withPrim primNe t nx ny) tis
```
Comment (jystic) [inline src/Icicle/Avalanche/Statement/Simp/Constructor.hs]: Would this be the same as using the generalised version below for Eq/Ne ?

Comment (tranma) [inline src/Icicle/Avalanche/Statement/Simp/Constructor.hs]: yea I guess it would be

Comment (jystic) [inline src/Icicle/Avalanche/Statement/Simp/Constructor.hs]: I think we should do that then :)

Comment (tranma) [inline src/Icicle/Avalanche/Statement/Simp/Constructor.hs]: we can do that for Eq but not Ne

Comment (jystic) [inline src/Icicle/Avalanche/Statement/Simp/Constructor.hs]: Not sure if it's better, but I guess we could have a function for Op -> AppendOp

**Comment by tranma** [comment]
you think this is ok, @amosr?

**Comment by jystic** [comment]
Looks great :space_invader: 

Now our comparison ops are superior to C! :cake:

**Comment by amosr** [comment]
woohoo!

**Pull request merged**